.github/workflows/engine_unit_test.yml

```diff
@@ -0,0 +1,57 @@
+name: Engine Unit Test
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on pull request events but only for the engine base branch and with changes in the src/engine/ directory.
+  pull_request:
+    types:
+      - ready_for_review
+      - synchronize
+    branches:
+      - 11334-dev-new-wazuh-engine
+    paths:
+      - 'src/engine/**'
+
+  # Allows to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Ensures only one instance of this workflow is running per PR
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  #cancel-in-progress: true
+
+env:
+  BUILD_TYPE: Release
+  ENGINE_DIR: ${{github.workspace}}/src/engine
+
+jobs:
+  build:
+    name: Engine Unit Test
+
+    # Runs only if the PR status is different to Draft
+    if: ${{ !github.event.pull_request.draft }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v3
+
+    - name: Cache CMake and ccache
+      uses: hendrikmuhs/ccache-action@v1.2
+      with:
+        key: ${{ github.workflow }}-${{ runner.os }}
+
+    - name: Configure CMake
+      # Configure the CMake build system with the specified build type
+      run: cmake --no-warn-unused-cli -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -S ${{env.ENGINE_DIR}} -B ${{env.ENGINE_DIR}}/build
+
+    - name: Build
+      # Build the specified target using CMake
+      run: cmake --build ${{env.ENGINE_DIR}}/build --config ${{env.BUILD_TYPE}} --target all -j2
```

Comment:
can we increase the number of parallel jobs?

Reply:
No, because it is restricted by the VM provided by Github which is dual core.